### PR TITLE
add pyramids package

### PR DESCRIPTION
**Pull request by MAfarrag**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by MAfarrag** [comment]
@conda-forge-linter @conda-forge/core @conda-forge/cf-python please check the win_64 pipeline i believe the error has nothing to do with my package but the installation of conda version

File: recipes/pyramids/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+test:
+  imports:
+    - pyramids
+
+about:
+  home: https://github.com/MAfarrag/pyramids
+  license: GPL3
+  license_file: LICENSE.md
+  license_family: GPL3
+  license_file: LICENSE.md
```
Comment (hmaarrfk) [inline recipes/pyramids/meta.yaml:48]: why is the license key duplicated?

File: recipes/pyramids/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  url: https://github.com/MAfarrag/pyramids/archive/{{ version }}.tar.gz
+  sha256: 7759146e4642f84c87c1996bfc21a72209e0bd0e994a521a2bce3cddbceff7a3
+
+build:
+  number: 0
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+
```
Comment (hmaarrfk) [inline recipes/pyramids/meta.yaml:18]:
```suggestion
```
please remove this blank line.

File: recipes/pyramids/LICENSE.md [hunk 3]
```diff
@@ -0,0 +1,675 @@
+### GNU GENERAL PUBLIC LICENSE
```
Comment (hmaarrfk) [inline recipes/pyramids/LICENSE.md:1]: Is it possible for you to create a new release with the license file included in the source dist? I think you need to add the following `metadata` key in `setup.cfg`
https://github.com/scikit-image/scikit-image/blob/main/setup.cfg#L10

**Review by hmaarrfk — COMMENTED** [review]
can you please revert the file permission changes you made to the azure-pipelines.


For what it s worth, noarch packages only need to pass on linux. The entry-points created on windows cause problems for the linter.

File: recipes/pyramids/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  version: {{ version }}
+
+source:
+  url: https://github.com/MAfarrag/pyramids/archive/{{ version }}.tar.gz
+  sha256: 7759146e4642f84c87c1996bfc21a72209e0bd0e994a521a2bce3cddbceff7a3
+
+build:
+  number: 0
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
```
Comment (hmaarrfk) [inline recipes/pyramids/meta.yaml:15]: You more or less need to copy entry points:
https://github.com/MAfarrag/pyramids/blob/main/setup.py#L32

For example:
https://github.com/conda-forge/pytest-feedstock/blob/main/recipe/meta.yaml#L16

File: recipes/pyramids/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    - pytest-cov >=2.12.1
+    - pytest >=6.2.5
+test:
+  imports:
+    - pyramids
+
+about:
+  home: https://github.com/MAfarrag/pyramids
+  license: GPL3
+  license_family: GPL3
```
Comment (hmaarrfk) [inline recipes/pyramids/meta.yaml:45]: license family should likely be `GPL`

license should be something from the list: https://spdx.org/licenses/

I'm not sure why the bot didn't flag the `license` as not being on the list.

**Comment by MAfarrag** [comment]
@hmaarrfk can you check it now all checks are passed

File: recipes/pyramids/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    - geoplot >=0.5.1
+    - pytest-cov >=2.12.1
+    - pytest >=6.2.5
+test:
+  imports:
+    - pyramids
+
+about:
+  home: https://github.com/MAfarrag/pyramids
+  license: GPL3
```
Comment (hmaarrfk) [inline recipes/pyramids/meta.yaml:44]: Repeating myself a little bit, the `license` needs to be something found in `Identifier` column exactly:
image

https://spdx.org/licenses/

File: recipes/pyramids/meta.yaml (hunk 4 above)
Comment (hmaarrfk) [reply recipes/pyramids/meta.yaml:15]: I don't understand why you makred this as resolved, even though you did not add any entry points.

**Comment by hmaarrfk** [comment]
please address the both's comment regarding matplotlib and my comment regarding entry points. Then ping me again.

**Comment by MAfarrag** [comment]
@hmaarrfk I have removed the cli from the package, for now, so no need for the entry point at all and linter comments are all done

**Comment by hmaarrfk** [comment]
Thank you for being explicit.

For future reference, when you do ask for help, and feedback is provided, it really helps if you explain what you did to address the feedback.

**Comment by hmaarrfk** [comment]
I'm really sorry to ask, but can you merge from conda-forge main and updatE? it seemed that you were 32k commits behind. This repo moves quickly, but to ensure good merges, I had to update. I think if you call `git merge upstream/main` it will work. 

Also, as a note, it is ill-advised to work off `main` or `master` even in your own fork.

**Comment by MAfarrag** [comment]
@hmaarrfk I have merged the main repo to my fork now it is not behind, and all checks have passed.

**Comment by hmaarrfk** [comment]
thanks!

**Pull request merged**